src/components/landing-page/LandingPage.tsx

```diff
@@ -9,6 +9,31 @@ import { useAppDispatch, useAppSelector } from '../../state/Hooks';
 
 
 export default function LandingPage() {
+	const query = new URLSearchParams(window.location.search);
```

Comment:
In React, this function is run every time an event occurs that would re-render the page. To prevent unnecessary re-runs, you should wrap it in a `useEffect(yourFunction, [])` hook. As a convention, I always put effects below the other hooks and component variables.

This will also solve an additional problem: conditional logic in a component should not occur before any React hooks. I'm not sure why this is a problem, but the React team claims it causes problems.